diff --git a/drivers/usb/gadget/function/f_uvc.c b/drivers/usb/gadget/function/f_uvc.c
index 5c850be..5a9ec10 100644
--- a/drivers/usb/gadget/function/f_uvc.c
+++ b/drivers/usb/gadget/function/f_uvc.c
@@ -802,7 +802,7 @@ static struct usb_function_instance *uvc_alloc_inst(void)
        cd->wObjectiveFocalLengthMax    = cpu_to_le16(0);
        cd->wOcularFocalLength          = cpu_to_le16(0);
        cd->bControlSize                = 3;
-       cd->bmControls[0]               = 2;
+       cd->bmControls[0]               = 14;
        cd->bmControls[1]               = 0;
        cd->bmControls[2]               = 0;
 
@@ -815,7 +815,7 @@ static struct usb_function_instance *uvc_alloc_inst(void)
        pd->wMaxMultiplier              = cpu_to_le16(16*1024);
        pd->bControlSize                = 2;
        pd->bmControls[0]               = 255;
-       pd->bmControls[1]               = 6;
+       pd->bmControls[1]               = 47;
        pd->iProcessing                 = 0;
 
        od = &opts->uvc_output_terminal;
